src/project_types/project.rs

```diff
@@ -154,20 +168,31 @@ impl Project {
         add: bool
     ) -> Result<(), Error> {
     
-        for mut u in &mut self.users {
-            if u.name == name {
-                if add {
-                    u.add_permission(perm);
-                } else {
-                    u.remove_permission(perm);
+        for i in 0..self.users.len() {
```

Comment:
Will do as above, but for the record, 0..v.len() is correct. the X..Y syntax creates a range, which acts like an iterator (didn't check if it actually was one).